Update "inquirer" to v1.0.2

Turbo87 opened the pull request:
Now that SBoudrias/Inquirer.js#362 is fixed we can update our "inquirer" dependency to the v1.x release series.

stefanpenner commented on lib/ui/index.js:
```diff
@@ -194,11 +193,13 @@ UI.prototype.prompt = function(questions, callback) {
+    inquirer.prompt(questions).then(function(result) {
+      callback(result);
+    }).catch(function() {
+      callback(null);
```
do we really want to catch and swallow all exceptions?

Turbo87 replied:
this is essentially just mirroring the previous behavior, but I'm open to other ideas.

stefanpenner replied:
i think we should rethrow / crash if there is an exception. No?

Turbo87 replied:
I tend to agree, I'm just not sure *how* to rethrow in this case. The old `callback` argument is not using the node `function(err, result)` signature so we can't pass any errors to the callback like that.

Turbo87 replied:
@stefanpenner any ideas?

twokul replied:
@Turbo87 @stefanpenner it seems with new `1.0.3` release, we should be able to re-throw ([commit link](https://github.com/SBoudrias/Inquirer.js/commit/19f5e1a4cadb1cc0c73f54c68691f9b61d241f0a)).

so then something like this should work:

```js
function rethrow(reason) {
  setTimeout(function() {
    throw reason;
  }, 0);
}
```

Turbo87 commented:
@stefanpenner apparently I missed that inquirer is still supporting the old callback interface too. this means that inquirer now has the exact same interface as our `prompt()` method so we can just pass along the arguments and the return value. I updated the PR to that and added some type info for the API docs.

stefanpenner commented:
@Turbo87 cool, seems like the tests are unhappy. Am i blocking something or ...

Turbo87 commented:
@stefanpenner looks  like AppVeyor is somewhat flaky at the moment, but Travis seems fine with these changes

stefanpenner commented:
@Turbo87 :shipit:

stefanpenner commented:
@homu r+

homu commented:
:pushpin: Commit c766d89 has been approved by `stefanpenner`

<!-- @homu r=stefanpenner c766d89f50625d253ac2a3b2d18ea1e78d0f3dff -->

Turbo87 commented:
@homu are you going to test this any time soon?!

stefanpenner commented:
i <3 @homu but its having some troubles lately...
